lib/src/dom/within.dart

```diff
@@ -50,12 +48,12 @@ class WithinQueries extends ScopedQueries {
 /// > See: <https://testing-library.com/docs/dom-testing-library/api-within>
 ///
 /// {@category Queries}
-WithinQueries within(Node node) {
+WithinQueries within(Node? node) {
   if (node == null) {
     throw ArgumentError.notNull('node');
   }
```

Comment:
I could make `node` non-nullable because it throws an error if it's null anyway, but then you wouldn't be able to do `within(ref.current)` without having to use the `!` operator - thoughts?

Reply:
I think if it throws for null, it'd be better to make it non-nullable, so that the null-check isn't hidden from the consumer.

The only places I'm seeing the `within(ref.current)` pattern are in RTL's `test/unit/dom/queries/by_role_test.dart` and not anywhere else, so making it non-nullable hopefully shouldn't have too much impact.

Reply:
Oh nice! Okay I'll update that!